File: web-client/src/domain/auth/CurrentSession/Session/entities/PersonalUserInfo.ts
import UserId from "../../../values/UserId";
import Email from "../../../values/Email";
import FullName from "../FullName";
import {PhotoPath} from "../../../values/PhotoPath";

class PersonalUserInfo {
    public constructor(
        public readonly id: UserId,
        public readonly email: Email,
        public readonly fullName: FullName,
        public readonly photo: PhotoPath | null
    ) {
    }
}

export default PersonalUserInfo;

File: backend/users-ms/src/main/java/com/azat4dev/booking/users/queries/domain/entities/PersonalUserInfo.java
package com.azat4dev.booking.users.queries.domain.entities;

import com.azat4dev.booking.shared.domain.values.user.UserId;
import com.azat4dev.booking.users.commands.domain.core.values.user.FullName;

import java.util.Optional;


public record PersonalUserInfo(
    UserId id,
    String email,
    EmailVerificationStatus emailVerificationStatus,
    FullName fullName,
    Optional<UserPhoto> photo
) {

    public enum EmailVerificationStatus {
        NOT_VERIFIED,
        VERIFIED
    }
}
